src/components/elements/annotationSidePanel/components/annotationWizard/steps/NewInstance.tsx

```diff
@@ -0,0 +1,166 @@
+
+/* Import Hooks */
+import { useTrigger } from 'app/Hooks';
+
+/* Import Types */
+import { DigitalSpecimen } from 'app/types/DigitalSpecimen';
+import { DigitalMedia } from 'app/types/DigitalMedia';
+import { AnnotationTarget, Dict, ParentClass } from 'app/Types';
+
+/* Import Components */
+import ParentClassification from './ParentClassification';
+import { Button } from 'components/elements/customUI/CustomUI';
+
+
+/* Props Type */
+type Props = {
+    superClass: DigitalSpecimen | DigitalMedia | Dict,
+    selected: boolean,
+    annotationTarget?: AnnotationTarget,
+    formValues?: Dict,
+    SetFieldValue?: Function
+};
+
+
+/**
+ * Component that renders a new, annotatable instance of the selected annotation target in the select instance step of the wizard
+ * @param superClass The provided super class
+ * @param selected Boolean indicating if the new instance option has been selected
+ * @param annotationTarget The annotation target set
+ * @param formValues The current values in the annotation wizard form
+ * @param SetFieldValue Function to set a value of a field in the form
+ * @returns JSX Component
+ */
+const NewInstance = (props: Props) => {
+    const { superClass, selected, annotationTarget, formValues, SetFieldValue } = props;
+
+    /* Hooks */
+    const trigger = useTrigger();
+
+    /* Base variables */
+    const [parentClasses, setParentClasses] = useState<ParentClass[]>([]);
+
+    /* Check for parent classes, they either contain the term 'has' or start with a capital */
+    trigger.SetTrigger(() => {
+        const parentClasses: ParentClass[] = [];
+
+        if (annotationTarget) {
+            const pathArray: string[] = ['$'];
+            const parents = jp.parse(annotationTarget.jsonPath).slice(1, -1);
+
+            parents.forEach((parent, index) => {
+                /* Filter path from schemas */
+                let parentName = parent.expression.value.replace('ods:', '').replace('dwc:', '').replace('dcterms:', '');
+
+                /* Add to pathArray */
+                pathArray.push(parent.expression.value);
+
+                if (parentName.includes('has') || parentName[0] === parentName[0].toUpperCase()) {
+                    /* If a parent class is empty, it needs to be targeted, if it is not completely empty, check for indexes */
+                    const parentNodes = jp.nodes(superClass, jp.stringify(pathArray).replace('][', ']..[').replaceAll('"', "'"));
+
+                    if (!parentNodes.length) {
+                        parentClasses.push({
+                            jsonPath: jp.stringify(pathArray).replaceAll('"', "'"),
+                            name: MakeJsonPathReadableString(jp.stringify(pathArray)),
+                            ...(index > 0 && { parentName: pathArray[index] }),
+                            present: false
+                        });
+                    }
+
+                    parentNodes.forEach((parentNode, parentIndex) => {
```

Comment:
you have a nested loop here, do you think you could build parentClasses outside the loop?

Reply:
```
const newParentClasses = [];
parentNodes.forEach((parentNode, parentIndex) => {
    newParentClasses.push({
        /*...populate object...*/
    });
});
setParentClasses(newParentClasses)
```

Reply:
There is a reason for this and that is because there can be multiple of the same parent, aka multiple events if you target the location class. Also, the JSON path selecter nodes function you see being used returns that level of of an array, containing indexes for all 'hits' (found indexes), which contains another array with all the instances themselves, so events for example. Hence the double loop